include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -52,6 +52,10 @@ class binary_reader
         assert(ia);
     }
 
+    // delete because of pointer members
+    binary_reader(binary_reader&) = delete;
+    binary_reader& operator=(binary_reader&) = delete;
+
```

Comment:
Caveat: I'm not up to speed on this file's implementation :)

Two things:
1. The comment doesn't feel right. This isn't non-copyable because it has pointers (that justification would only apply if the pointers were *owned* by this class [aka, the destructor deletes them], but this class has no destructor).  Isn't it more that it "consumes" the inputs and expects that it has exclusive operation on the input stream. Ie, if you copy this class and then interleave calls to each copy, there is nothing coherent that comes from that. Therefore there is an implication of "exclusive access", not that there are pointers.
2. Your patch changes the ABI here because declaring the copy constructors removes the move constructors, and I believe this class should still be movable. Since this patch is all about following Effective rules, we should follow Rule of 5 and declare all the special functions explicitly here.